client/state/ui/notices/actions.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * Internal dependencies
+ */
+import {
+	NEW_NOTICE,
+	REMOVE_NOTICE
+} from 'state/action-types';
+
+import { uniqueId } from 'lodash';
+
+export function createNoticeAction( status, text, options = {} ) {
+	return {
+		noticeId: uniqueId(),
+		duration: options.duration,
+		showDismiss: ( typeof options.showDismiss === 'boolean' ? options.showDismiss : true ),
+		type: NEW_NOTICE,
+		status: status,
+		text: text
+	};
+}
+
+export function removeNoticeAction( noticeId ) {
+	return {
+		noticeId: noticeId,
+		type: REMOVE_NOTICE
+	};
+}
+
+export function noticesMapDispatchToProps( dispatch ) {
```

Comment:
>But then again it is not a mixin and there is no way to turn it into one.

Depends on the definition of mixin :smiley: It does appear that the majority of what we have there are "React mixins", though the [`emitter` mixin](https://github.com/Automattic/wp-calypso/tree/master/shared/lib/mixins/emitter) is one exception. An alternative is to have a directory in `lib` containing these types of helpers.

>And we need this helper, it will have ~20 lines of code we would have to copy across 60+ files

That seems a bit of a stretch, and I still question whether we need these helpers. Sure, what we have here is 20 lines, but I still think this could be simplified to a one-line function in the `connect` binding:

```jsx
function Button( { createSuccessNotice } ) {
	return (
		<button onClick={ () => createSuccessNotice( 'Great success!' ) }>
			Create Success
		</button>
	);
}

export default connect( null, ( dispatch ) => {
	return bindActionCreators( { createSuccessNotice }, dispatch );
} )( Button );
```